core/src/com/unciv/ui/worldscreen/mainmenu/OptionsPopup.kt

```diff
@@ -962,20 +969,30 @@ class OptionsPopup(
         add(checkbox).colspan(2).left().row()
     }
 
-    private fun formatMultiplayerUrlInput(input: String): String {
-        var result : String
+    private fun fixTextFieldUrlOnType(TextField: TextField) {
+        var text: String = TextField.text
+        var cursor: Int = minOf(TextField.cursorPosition, text.length)
 
-        // remove all whitespaces
-        result = Regex("\\s+").replace(input, "")
+        // if text is 'http:' or 'https:' auto append '//'
+        if (Regex("^https?:$").containsMatchIn(text)) {
+            TextField.appendText("//")
+            return
+        }
+
+        val textBeforeCursor: String = text.substring(0, cursor)
 
         // replace multiple slash with a single one
-        result = Regex("/{2,}").replace(result, "/")
+        val multipleSlashes = Regex("(?<!(^https?:))/{2,}")
```

Comment:
Add a regexplaination - I'm no stranger to regex and I don't currently trust what's going here

Reply:
`(?<!pattern)` is [Negative Lookbehind](https://www.regextutorial.org/positive-and-negative-lookbehind-assertions.php)
It matches something if that this is not immediately after a pattern. For example `(?<!a)m` matches m not followed by a. For example in `ampm` it matches the m after p but not the m after a. https://regexr.com/6llgv
Since explaination would get too big I just committed the previous approach.